pkg/packagekit/autoupdates.jsx

```diff
@@ -249,67 +248,128 @@ function getBackend(forceReinit) {
-        this.setState({ pending: true, pendingEnable: enabled });
-        this.debouncedSetConfig(enabled, type, day, time);
+        this.setState({ pending: true });
+        backend.setConfig(enabled, type, day, time)
+                .always((b) => {
+                    this.debugBackendState("handleChange: setConfig finished");
+                    if (this.props.onInitialized) {
+                        this.props.onInitialized({
+                            autoUpdatesEnabled: enabled,
+                            autoUpdatesType: type,
+                            autoUpdatesDay: day,
+                            autoUpdatesTime: time,
+                        });
+                    }
+                    this.setState({ pending: false, showModal: false });
+                });
     }
 
     render() {
-        const backend = this.state.backend;
-        if (!backend)
+        if (!this.state.backend)
             return null;
 
-        let autoConfig;
         const enabled = !this.state.pending && this.props.privileged;
-
-        if (backend.enabled && backend.installed) {
-            const hours = Array.from(Array(24).keys());
-
-            autoConfig = (
-                <div className="auto-conf">
-                    <span className="auto-conf-group">
-                        <Select.Select id="auto-update-type" enabled={enabled} initial={backend.type}
-                                       onChange={ t => this.handleChange(null, t, null, null) }>
-                            <Select.SelectEntry data="all">{_("Apply all updates")}</Select.SelectEntry>
-                            <Select.SelectEntry data="security">{_("Apply security updates")}</Select.SelectEntry>
-                        </Select.Select>
-                    </span>
-
-                    <span className="auto-conf-group">
-                        <Select.Select id="auto-update-day" enabled={enabled} initial={backend.day}
-                                       onChange={ d => this.handleChange(null, null, d, null) }>
+        const hours = Array.from(Array(24).keys());
+        const body = (
+            <Form isHorizontal>
+                <FormGroup fieldId="type" label={_("Type")}>
+                    <Radio isChecked={!this.state.enabled}
+                           onChange={e => this.setState({ enabled: false, type: null })}
+                           enabled={enabled}
+                           label={_("No updates")}
+                           id="no-updates"
+                           name="radio-no-updates" />
+                    <Radio isChecked={this.state.enabled && this.state.type === "security"}
+                           onChange={e => this.setState({ enabled: true, type: "security" })}
+                           enabled={enabled}
+                           label={_("Security updates only")}
+                           id="security-updates"
+                           name="radio-security-updates" />
```

Comment:
The name field of the radio buttons that belong to one group should be the same. See https://www.w3schools.com/tags/att_input_type_radio.asp

Reply:
Fixed